block/local/adapter.go

```diff
@@ -127,7 +140,28 @@ func (l *Adapter) Remove(obj block.ObjectPointer) error {
 		return err
 	}
 	p = filepath.Clean(p)
-	return os.Remove(p)
+	err = os.Remove(p)
+	if err != nil {
+		return err
+	}
+	if l.removeEmptyDir {
+		dir := filepath.Dir(p)
+		removeEmptyDirUntil(dir, l.path)
+	}
+	return nil
+}
+
+func removeEmptyDirUntil(dir string, stopAt string) {
+	for {
+		err := os.Remove(dir)
+		if err != nil {
+			break
+		}
+		dir = filepath.Dir(dir)
+		if dir == "/" || strings.HasPrefix(stopAt, dir) {
```

Comment:
This tests whether `stopAt` begins with `dir` (https://golang.org/pkg/strings/#HasPrefix).  Given that this is supposed to protect files that are outside the adapter's area of responsibility it might be safer to say at the top of the loop
```go
for dir != "" && strings.HasPrefix(dir, stopAt)
```

The differences are in what happens when the initial condition `strings.HasPrefix(dir, stopAt)` doesn't hold when the function is called (we don't want the loop to go all the way to the root).

Reply:
Agree, at first, I wanted to support the case there is no stop and I just need to remove the folder that holds the file.

Reply:
I have to test using `!strings.HasPrefix(stopAt, dir)` - don't want to test the case of the adapter root folder